fedimint-bitcoind/src/bitcoincore_rpc.rs

```diff
@@ -292,3 +299,112 @@ impl IBitcoindRpc for ElectrumClient {
         })
     }
 }
+
+// TODO: (@oleonardolima) Should both this EsploraClient and ElectrumClient be extracted to different files/namespaces ?
+pub struct EsploraClient(esplora_client::AsyncClient);
+
+impl EsploraClient {
+    fn new(url: &Url) -> anyhow::Result<Self> {
+        let builder = esplora_client::Builder::new(url.as_str());
+        let client = builder.build_async()?;
+        Ok(Self(client))
+    }
+}
+
+impl fmt::Debug for EsploraClient {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("EsploraClient")
+    }
+}
+
+#[async_trait]
+impl IBitcoindRpc for EsploraClient {
+    fn backend_type(&self) -> BitcoinRpcBackendType {
+        BitcoinRpcBackendType::Esplora
+    }
+
+    async fn get_network(&self) -> Result<Network> {
+        let genesis_height: u32 = 0;
+        let genesis_hash = self.0.get_block_hash(genesis_height).await?;
+
+        // TODO: How could we extract the BlockHash to a constant, and use something like the `bitcoin::blockdata::constants::genesis_block` from rust-bitcoin ?
```

Comment:
`const MAINNET_GENESIS_BLOCK_HEX : &'str = "..."`? Seems like a good idea to share. If `rust-bitcoin` has one, even better.

Reply:
I know that rust-bitcoin exposes the fn based on the `Network` as a parameter, returning the `Block` 🤔.
I'll double-check if rust-bitcoin exposes its constants, otherwise, I'll add the constants.